### PR TITLE
[AI-119] Add Fiona /fiona slash command help and telemetry

### PR DESCRIPTION
**Pull request by roberthunterjr**
## Summary
- Adds a new `/fiona` slash command to the Slack app manifest with usage hint support.
- Registers a dedicated command listener with explicit routing for `help`, `ask`, `search`, and unknown sub-commands.
- `/fiona help` (and bare `/fiona`) returns an immediate ephemeral help response and records `slash_help` telemetry without invoking the LLM.
- `/fiona ask` and `/fiona search` return an explicit &#34;not yet available&#34; response and record `slash_ask` / `slash_search` telemetry respectively.
- Unknown sub-commands log a warning, fall back to the help response, and record `slash_help` telemetry.
- All handlers wrap `ack()` in a try/catch to prevent Slack `dispatch_failed`; missing required fields (`user_id`, `channel_id`, `trigger_id`) skip the Cosmos write with a logged warning.
- Adds focused unit tests for all routing paths, ack failure resilience, missing fields guard, and telemetry for each interaction type.

## Walkthrough: verify `/fiona help` in Slack (Sandbox)
1. Update/reinstall the Fiona Slack app in your workspace so the latest manifest is applied.
2. Open Slack **Sandbox** and go to **Apps**.
3. In the agent selector/app list, choose **Fiona** with the **black dots icon** (not `Fiona (Robert)`).
4. Open a channel or DM with that Fiona app and run `/fiona help`.
5. Confirm you receive an ephemeral help response showing:
   - `/fiona help`
   - `/fiona ask ` (coming soon)
   - `/fiona search ` (coming soon)
6. Run `/fiona ask` or `/fiona search` and confirm you receive a &#34;not yet available&#34; message (not the full help menu).
7. Run `/fiona foo` and confirm it falls back to the help response.

### Sandbox selector reference
image

### Slash command in action
image

## Testing notes
- Latest telemetry/schema updates were tested against the **Fiona Insiders** bot in the **Fiona Sandbox** workspace.

### Cosmos telemetry evidence (Sandbox / Insiders)
image

### CODE REVIEW

File: apps/fiona-slack/src/listeners/commands/fiona.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  // Slash commands have no thread_ts or message_ts; trigger_id is unique per
+  // invocation and serves as both identifiers for the Cosmos document ID.
+  recordInteraction({
+    userId: user_id,
+    teamId: team_id,
+    channelId: channel_id,
+    threadTs: trigger_id,
+    messageTs: trigger_id,
+    interactionType: 'slash_help',
+    status: 'success',
```
[the remark this answers is not in the document]
Comment (roberthunterjr) [reply apps/fiona-slack/src/listeners/commands/fiona.js]: Addressed in latest commit on this PR branch: updated interaction-store JSDoc to explicitly support interactionType='slash_help' and trigger_id-backed threadTs/messageTs identifiers, added a regression test for slash_help payloads, and updated usage analytics/reporting documentation so downstream session/dashboard expectations describe these identifiers explicitly.

File: docs/usage-report/2026-03-20-fiona-slack-usage-analytics-plan.md
Comment (stephenfuqua) [inline docs/usage-report/2026-03-20-fiona-slack-usage-analytics-plan.md:1]: @roberthunterjr suggestion: this is a good example of why we shouldn't have the two files (which in this case was my doing!). There is no need to maintain this plan document. 

No concern about it in this PR, but perhaps I should do a separate PR to clean up these usage-report files.

**Comment by stephenfuqua** [comment]
# PR #52 Review — [AI-119] Add Fiona /fiona slash command help and telemetry
> Issue AI-119: /fiona help slash command
> Reviewed by: Security · Functionality · Maintainability · Usability · Test Coverage

---

## Critical

- **[Security + Functionality]** `manifest.json` is malformed — missing closing `}` and all required Slack fields
  The `slash_commands` array entry is never closed (`}` missing before `]`), making the file invalid JSON. Slack also requires `command` and `description` fields at minimum; without `command: "/fiona"`, Slack will reject the manifest on upload and the slash command will never be registered. This is a deployment blocker.
  Fix:
  ```json
  "slash_commands": [
    {
      "command": "/fiona",
      "description": "Interact with Fiona, your Ed-Fi AI assistant",
      "usage_hint": "[help]",
      "should_escape": false
    }
  ]
  ```
  Note: `should_escape: true` (the only field currently present) would URL-encode user/channel mentions in `command.text`, corrupting sub-command parsing. Use `false`.

## High

- **[Security]** `trigger_id` used as both `threadTs` and `messageTs` yields a semantically odd Cosmos document ID.
  `trigger_id` is a short-lived opaque Slack token. Using it for both fields produces IDs like `U12345_trigger-abc_trigger-abc` — the duplicate segments are redundant, the 3-second TTL makes the value meaningless as a session identifier, and the `upsert` call means a collision would silently overwrite a record with no error surfaced.
  Fix: Use `trigger_id` for `messageTs` and either a timestamp+UUID for `threadTs`, or document explicitly why the dual-mapping is acceptable.

- **[Maintainability]** Unknown sub-commands fall through to help silently — no log line, no telemetry distinction.
  `case '':` and `default:` both call `handleHelp` with identical `interactionType: 'slash_help'` and no logging. In production it will be impossible to distinguish a bare `/fiona` from `/fiona typo`. When `ask` and `search` are promoted to real handlers, the current fallthrough structure makes the switch dangerous to extend.
  Fix: Add a log line in `default`: `logger?.info?.(...)` and consider a distinct `interactionType` (e.g., `'slash_unknown'`) for the fallback.

## Medium

- **[Security]** `err.message` from Cosmos write failure logged verbatim at warn — may surface internal infrastructure details (endpoint URLs, connection strings).
  Fix: Log only `err.name` at warn; full `err` at debug only.

- **[Security]** No guard for missing `command` fields before passing to `recordInteraction`.
  Fix: Add explicit presence check for `user_id` and `trigger_id` in `handleHelp` before calling `recordInteraction`.

- **[Usability]** "Coming soon" sub-commands advertised in help text, but when typed the user silently gets the help message again with no explanation.
  Fix: Either remove `/fiona ask` and `/fiona search` from `HELP_TEXT` until implemented, or add explicit `case 'ask':` / `case 'search':` handlers that respond with a clear "not yet available" message pointing users to @-mention/DM.

- **[Usability]** No error boundary around `ack()` — if it throws, the user sees nothing and Slack shows `dispatch_failed`.
  Fix: Wrap `fionaCommandCallback` in a try/catch that calls bare `ack()` on error.

- **[Maintainability]** `trigger_id` dual-mapped to `threadTs`/`messageTs` with no guard or adapter abstraction.
  Fix: Add a guard for missing `trigger_id`; extract a `slashCommandToInteractionIds(command)` helper so future sub-commands share the mapping.

- **[Maintainability]** `HELP_TEXT` "coming soon" entries are coupled to the test's "unknown sub-command" `it.each` list.
  When `ask`/`search` are implemented, both must be updated in sync or a misleading test failure results.
  Fix: Add a comment in the test noting these are temporarily in the fallback list.

## Low

- **[Security]** `commands` OAuth scope in `oauth_config` is unnecessary — slash commands are enabled via the manifest entry, not an OAuth scope.
  Fix: Remove `"commands"` from `oauth_config.scopes.bot`.

- **[Functionality]** No test for `command.text` being `null` or `undefined` (implementation handles it correctly via `??`, but it is unexercised).

- **[Usability]** "Coming soon" label in help text gives no alternative action for the user.
  Fix: Replace with `(not yet available — use @-mention or DM instead)`.

- **[Usability]** `/fiona help` is not mentioned in the assistant thread greeting, limiting discoverability.
  Fix: Append discovery hint to `say('Hi, how can I help?')`.

- **[Maintainability]** `fionaCommandCallback` logs nothing on the happy path — zero operational trace if Cosmos is not configured.
  Fix: Add an info log at the top of the callback.

- **[Test]** "No LLM call" guard only intercepts static imports; dynamic `import()` would bypass it. Add a comment acknowledging the scope.

- **[Test]** Single `await Promise.resolve()` flush is fragile for fire-and-forget assertions.
  Fix: Use `await new Promise(setImmediate)` or two consecutive flushes.

- **[Test]** `ack()` ephemeral semantics are implicit — add a comment explaining that `ack(string)` is how Bolt sends ephemeral slash command replies.

## Nit

- `switch` with all three cases falling through to the same handler — restructure as `if/else` when real sub-commands land.
- Resilience test's `stringContaining('cosmos down')` couples to exact error message text; `stringContaining('Failed to record')` is more stable.
- `mockLogger` includes unused `info` and `error` mock methods in `beforeEach`.
- `:wave:` emoji has no alt text on some Slack surfaces (minor a11y awareness, no action required).

---

## Specialist Verdicts

| Reviewer | Verdict |
|----------|---------|
| Security | No runtime exploits — Bolt's HMAC verification is the correct trust boundary — but the malformed manifest blocks deployment, `trigger_id` undermines telemetry integrity, and the `commands` scope unnecessarily widens the OAuth surface. |
| Functionality | **INCOMPLETE** — `manifest.json` is invalid JSON with a broken `slash_commands` entry missing required Slack fields; the slash command cannot be registered in Slack. |
| Maintainability | Core implementation is clean and fits the existing module structure well; main friction points are the silent fallthrough on unknown sub-commands and the `trigger_id` dual-mapping with no adapter or guard. |
| Usability | Solid for scope, but advertising two non-functional sub-commands creates a misleading affordance that will cause user confusion immediately. |
| Test Coverage | **STRONG** — all six acceptance criteria have corresponding tests; identified gaps are edge-case fragilities, not missing scenarios. |

---

## Summary

The **Critical blocker is `manifest.json`** — structurally broken JSON with missing required Slack fields means the `/fiona` command cannot be registered at all. Fix that first. Before merge, also address the "coming soon" sub-commands in `HELP_TEXT` — advertising `/fiona ask` and `/fiona search` while silently routing them to help will confuse users immediately. The core implementation (routing, telemetry, error handling, test suite) is well-executed.

---
_Review generated with Claude Code (five-specialist parallel review)_


[AI-119]: https://edfi.atlassian.net/browse/AI-119?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Review by stephenfuqua — APPROVED** [review]

**Comment by stephenfuqua** [comment]
Some low priority items _should_ be addressed for better usability, but we'll tackle them in a separate work item (yet to be created).
